R/kernel.r

```diff
@@ -242,6 +243,92 @@ complete = function(request) {
         cursor_end = start_position + nchar(c.info$token)))
 },
 
+inspect = function(request) {
+    make_section_header_plain <- function (txt) paste0("# ", txt, ":\n")
+    make_section_header_html <- function (txt) paste0("<h1>", txt, ":</h1>\n")
+    # 5.0 protocol:
+    code <- request$content$code
+    cursor_pos <- request$content$cursor_pos
+
+    # Get token under the cursor_pos.
+    # There may be better ways to do that than this.
+    token <- ""
+    for (i in seq(cursor_pos, nchar(code))) {
+        topic_candidate <- utils:::.guessTokenFromLine(code, i)
+        if (!grepl(token, topic_candidate)) break
+        token <- topic_candidate
+    }
+
+    if (nchar(token) == 0) {
+        data <- list()
+        found <- FALSE
+    } else {
+        data <- list()
+
+        class_data <- tryCatch(
+            eval(parse(text = paste0("class(", token, ")"))),
```

Comment:
all evals from this point on are unnecessary.

just `obj <- get(token)` should get you the object, then you can just call functions normally:

`class(obj)` works, `obj_help <- help(obj)` does as well, and then you can call `IRdisplay::prepare_mimebundle` on both `obj` and `obj_help` directly.